ni-oe-init-build-env

```diff
@@ -1,8 +1,51 @@
-TEMPLATECONF=${TEMPLATECONF:-$PWD/sources/meta-nilrt/conf}
-BITBAKEDIR=${BITBAKEDIR:-$PWD/sources/bitbake}
-BUILDDIR=${BUILDDIR:-$PWD/build}
-BB_ENV_EXTRAWHITE=${BB_ENV_EXTRAWHITE:-ENABLE_BUILD_TAG_PUSH}
+#!/usr/bin/env
+
+SCRIPT_ROOT=$(realpath $(dirname ${BASH_SOURCE:-${0}}))
+
+enable_ni_org_conf=false
+positionals=()
+
+while [ $# -ge 1 ]; do case "$1" in
+	-o|--org)
+		enable_ni_org_conf=true
+		shift
+		;;
+	*)
+		positionals+=($1)
+		shift
+		;;
+esac; done
+
+
+BITBAKEDIR=${BITBAKEDIR:-${SCRIPT_ROOT}/sources/bitbake}
+BUILDDIR=${BUILDDIR:-${SCRIPT_ROOT}/build}
+
+BB_ENV_EXTRAWHITE="${BB_ENV_EXTRAWHITE} ENABLE_BUILD_TAG_PUSH"
+
+# define the location of bitbake configuration files, which will be copied
+# into the build workspace, if one needs to be created.
+TEMPLATECONF=${TEMPLATECONF:-${SCRIPT_ROOT}/sources/meta-nilrt/conf}
 
-export PS1="(bb) $PS1"
 export TEMPLATECONF
-source $PWD/sources/openembedded-core/oe-init-build-env
+
+# Call OE-upstream's build env initialization script, which will create a build
+# workspace called either "${1:-build}/" and `cd` into it.
+cd ${SCRIPT_ROOT}
```

Comment:
Not 100% sure why the cd is necessary ... is it to cover the possibility of someone calling ni-oe-init-build-env without first being in the same directory as the script?

Reply:
It's been a while since I wrote that line, but iirc. it is to coerce the `build/` directory to be created within the top level of the nilrt repo. Our bitbake configurations don't behave predictably when the build workspace is in another location.